kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -0,0 +1,307 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiConsumer;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.IsolationLevel;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.jbpm.services.api.DeploymentEvent;
+import org.jbpm.services.api.DeploymentEventListener;
+import org.jbpm.services.api.DeploymentService;
+import org.jbpm.services.api.ListenerSupport;
+import org.jbpm.services.api.ProcessService;
+import org.jbpm.services.api.model.DeployedAsset;
+import org.jbpm.services.api.model.ProcessDefinition;
+import org.jbpm.services.api.model.SignalDesc;
+import org.jbpm.services.api.model.SignalType;
+import org.kie.server.api.KieServerConstants;
+import org.kie.server.services.api.KieContainerInstance;
+import org.kie.server.services.api.KieServerExtension;
+import org.kie.server.services.api.KieServerRegistry;
+import org.kie.server.services.api.SupportedTransports;
+import org.kie.server.services.impl.KieServerImpl;
+import org.kie.server.services.jbpm.JbpmKieServerExtension;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaServerExtension implements KieServerExtension, DeploymentEventListener, Runnable {
+
+    public static final String EXTENSION_NAME = "Kafka";
+    static final String KAFKA_EXTENSION_PREFIX = "org.kie.server.jbpm-kafka.ext.";
+    static final String TOPIC_PREFIX = KAFKA_EXTENSION_PREFIX + "topics.";
+    private static final Logger logger = LoggerFactory.getLogger(KafkaServerExtension.class);
+
+    private AtomicBoolean initialized = new AtomicBoolean();
+    private Consumer<String, byte[]> consumer;
+    private ListenerSupport deploymentService;
+    private ProcessService processService;
+    private ScheduledExecutorService pollService;
+    private ExecutorService notifyService;
+
+    private Map<String, SignalInfo> topic2Signal = new ConcurrentHashMap<>();
+
+    @Override
+    public boolean isInitialized() {
+        return initialized.get();
+    }
+
+    @Override
+    public boolean isActive() {
+        return !Boolean.getBoolean(KieServerConstants.KIE_KAFKA_SERVER_EXT_DISABLED);
+    }
+
+    @Override
+    public void init(KieServerImpl kieServer, KieServerRegistry registry) {
+        if (initialized.getAndSet(true)) {
+            return;
+        }
+        pollService = Executors.newSingleThreadScheduledExecutor();
+        notifyService = Executors.newCachedThreadPool();
```

Comment:
make this configurable (number of threads)

Reply:
I think the usage of newCachedThreadPool matches the use case, as per documentation
* Creates a thread pool that creates new threads as needed, but
     * will reuse previously constructed threads when they are
     * available.  These pools will typically improve the performance
     * of programs that execute many short-lived asynchronous tasks.
     * Calls to {@code execute} will reuse previously constructed
     * threads if available. If no existing thread is available, a new
     * thread will be created and added to the pool. Threads that have
     * not been used for sixty seconds are terminated and removed from
     * the cache. Thus, a pool that remains idle for long enough will
     * not consume any resources. Note that pools with similar
     * properties but different details (for example, timeout parameters)
     * may be created using {@link ThreadPoolExecutor} constructors.

Reply:
Upper limit to avoid thread exhaustion set